src/application_logic/storage/createItem.ts

```diff
@@ -0,0 +1,47 @@
+import { showTodoArea } from "components/showTodos/showTodos";
+import { Timestamp, addDoc } from "firebase/firestore";
+import { auth } from "../auth";
+import { ProjectData, TodoData, Project, Todo } from "types";
+import { projects, todosOfProject } from "./useDb";
+
+// factory for projects
+const createProjectData = (name: string, isInbox?: boolean): ProjectData => {
+  return {
+    name,
+    createdDate: Timestamp.now(),
+    ownerID: auth.currentUser?.uid!,
+    ...(isInbox && { isInbox: true }),
+    isArchived: false,
+  };
+};
+
+// factory for todos
+const createTodoData = (name: string): TodoData => {
+  return {
+    name,
+    complete: false,
+    description: "",
+    dueDate: null,
+    priority: 4,
+    createdDate: Timestamp.now(),
+    ownerID: auth.currentUser?.uid!,
+  };
+};
+
+const createItem = (type: "project" | "todo", parentProject?: Project) => {
```

Comment:
I haven't checked the whole source code, so I can't say for sure. But my guess is, that there probably isn't much benefit of merging those too functions together. If you split them up, you save an if/else statement and you can get rid of the type parameter. Also it becomes more precise what the function does.